scripts/kwarg_permissions.py

```diff
@@ -0,0 +1,115 @@
+from typing import Optional
+
+import libcst as cst
+import libcst.codemod.visitors as codevisitors
+from libcst import codemod
+
+from disnake import Permissions
+
+
+class PermissionTypings(codemod.VisitorBasedCodemodCommand):
+    DESCRIPTION: str = "Adds overloads for all permissions."
+
+    def __init__(self, context: codemod.CodemodContext):
+        super().__init__(context)
+        # add all of the permissions
+        self.permissions = sorted(Permissions.VALID_FLAGS.keys())
+
+    def transform_module(self, tree: cst.Module) -> cst.Module:
+        if "@_overload_with_permissions" not in tree.code:
+            raise codemod.SkipFile(
+                "this module does not contain the required decorator: `@_overload_with_permissions`."
+            )
+        return super().transform_module(tree)
+
+    def on_visit(self, node: cst.CSTNode) -> bool:
+        return super().on_visit(node)
```

Comment:
This is an unnecessary overload

Reply:
*function https://github.com/DisnakeDev/disnake/pull/708/commits/13b5ebfc8fff55dd9a707ad196050fa7f75a2377